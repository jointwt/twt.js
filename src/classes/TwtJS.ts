import methods from '../methods';

import {
  APIOptions,
  PostParams,
  Account,
  NewAccount,
  TimelineParams,
  UserOrFeed,
} from '../types.d';

export default class TwtJS {
  url: string;

  token: string;

  constructor(options: APIOptions) {
    this.url = options.url;
    this.token = options.token || '';
  }

  ping = () => methods.ping({
    url: this.url,
    token: this.token,
  });

  register = (account: NewAccount) => methods.register(account, {
    url: this.url,
    token: this.token,
  });

  authenticate = async (account: Account) => {
    const response = await methods.authenticate(account, {
      url: this.url,
    });
    this.token = (response.data as any).token;
    return this.token;
  };

  post = (params: PostParams) => methods.post(params, {
    url: this.url,
    token: this.token,
  });

  getTimeline = (params: TimelineParams) => methods.getTimeline(params, {
    url: this.url,
    token: this.token,
  });

  getDiscover = (params: TimelineParams) => methods.getDiscover(params, {
    url: this.url,
    token: this.token,
  });

  follow = (userOrFeed: UserOrFeed) => methods.follow(userOrFeed, {
    url: this.url,
    token: this.token,
  });
}
