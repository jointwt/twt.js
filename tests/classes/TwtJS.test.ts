import TwtJS from '../../src/classes/TwtJS';

jest.mock('../../src/methods/index', () => ({
  ping: () => Promise.resolve({
    message: 'successful mocked ping call',
  }),
  register: () => Promise.resolve({
    message: 'successful mocked register call',
  }),
  authenticate: () => Promise.resolve({
    data: {
      token: 'sample_mocked_token',
    },
    message: 'successful mocked authenticate call',
  }),
  post: () => Promise.resolve({
    message: 'successful mocked post call',
  }),
  getTimeline: () => Promise.resolve({
    message: 'successful mocked timeline call',
  }),
  getDiscover: () => Promise.resolve({
    message: 'successful mocked discover call',
  }),
  follow: () => Promise.resolve({
    message: 'successful mocked follow call',
  }),
}));

const getTwtInstance = () => new TwtJS({
  url: 'https://sample.url',
});

test('TwtJS class should be defined', () => {
  expect(TwtJS).toBeDefined();
});

test('TwtJS class should expose all API methods', () => {
  const twtJs = getTwtInstance();
  expect(typeof twtJs.ping).toBe('function');
  expect(typeof twtJs.register).toBe('function');
  expect(typeof twtJs.authenticate).toBe('function');
  expect(typeof twtJs.post).toBe('function');
  expect(typeof twtJs.getTimeline).toBe('function');
  expect(typeof twtJs.getDiscover).toBe('function');
  expect(typeof twtJs.follow).toBe('function');
});

test('ping should call methods.ping', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.ping();
  expect(data).toMatchSnapshot();
});

test('register should call methods.register', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.register({
    username: 'test_username',
    password: 'test_password',
    email: 'test@email.tld',
  });

  expect(data).toMatchSnapshot();
});

test('authenticate should call methods.authenticate', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.authenticate({
    username: 'test_username',
    password: 'test_password',
  });

  expect(data).toMatchSnapshot();
});

test('post should call methods.post', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.post({
    text: 'sample_text',
    postas: 'sample_user',
  });

  expect(data).toMatchSnapshot();
});


test('getTimeline should call methods.getTimeline', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.getTimeline({
    page: 1,
  });

  expect(data).toMatchSnapshot();
});

test('getTimeline should call methods.getTimeline', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.getTimeline({
    page: 1,
  });

  expect(data).toMatchSnapshot();
});

test('getDiscover should call methods.getDiscover', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.getDiscover({
    page: 1,
  });

  expect(data).toMatchSnapshot();
});

test('follow should call methods.follow', async () => {
  const twtJs = getTwtInstance();
  const data = await twtJs.follow({
    nick: 'sample_nick',
    url: 'https://sample_feed_url',
  });

  expect(data).toMatchSnapshot();
});