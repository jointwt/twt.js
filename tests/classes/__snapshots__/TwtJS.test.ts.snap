// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authenticate should call methods.authenticate 1`] = `"sample_mocked_token"`;

exports[`follow should call methods.follow 1`] = `
Object {
  "message": "successful mocked follow call",
}
`;

exports[`getDiscover should call methods.getDiscover 1`] = `
Object {
  "message": "successful mocked discover call",
}
`;

exports[`getTimeline should call methods.getTimeline 1`] = `
Object {
  "message": "successful mocked timeline call",
}
`;

exports[`getTimeline should call methods.getTimeline 2`] = `
Object {
  "message": "successful mocked timeline call",
}
`;

exports[`ping should call methods.ping 1`] = `
Object {
  "message": "successful mocked ping call",
}
`;

exports[`post should call methods.post 1`] = `
Object {
  "message": "successful mocked post call",
}
`;

exports[`register should call methods.register 1`] = `
Object {
  "message": "successful mocked register call",
}
`;
